File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q3-pascals-triangle-ii.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    let row = [1];
    for (let i = 1; i <= rowIndex; i++) {
        let newRow = [];
        newRow[0] = 1;
        newRow[i] = 1;
        for (let j = 1; j < i; j++) {
            newRow[j] = row[j - 1] + row[j];
        }
        row = newRow;
    }
    return row;
    
};

File: ELEC881_Tabnine/src/Original Prompt/Python/q3-pascals-triangle-ii.py
'''
Given an integer rowIndex, return the rowIndexth (0-indexed) row of the Pascal's triangle.

In Pascal's triangle, each number is the sum of the two numbers directly above it as shown:


 

Example 1:

Input: rowIndex = 3
Output: [1,3,3,1]
Example 2:

Input: rowIndex = 0
Output: [1]
Example 3:

Input: rowIndex = 1
Output: [1,1]
 

Constraints:

0 <= rowIndex <= 33
'''
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0:
            return [1]
        prevRow = [1]
        for i in range(1, rowIndex + 1):
            row = [1]
            for j in range(1, i):
                row.append(prevRow[j - 1] + prevRow[j])
            row.append(1)
            prevRow = row
        return row
        